src/extra/recipes/grating.py

```diff
@@ -122,45 +195,108 @@ def load_data(self):
                      grating=self._grating_signal,
                      )
         energy_ids = np.arange(len(self.calibration_energies))
+        # average data in each mono scan bin
         with ProcessPoolExecutor() as p:
             data = np.stack(list(p.map(fn, energy_ids)), axis=0)
-        self.calibration_data = data - self.bkg
+        # subtract the background
+        bkg_unc = np.zeros_like(data).mean(0)
+        if self.bkg is not None:
+            data = data - self.bkg
+            bkg_unc = self.bkg_unc
+        else:
+            self.bkg_unc = bkg_unc
+            self.bkg = np.zeros_like(data).mean(0)
+        self.calibration_unc = bkg_unc
+        # skip offset and collect pulse data each pulse_period samples only
+        self.calibration_data = data[:, self.offset::self.pulse_period, self.min_pixel:self.max_pixel]
+        self.calibration_unc = self.calibration_unc[self.offset::self.pulse_period, self.min_pixel:self.max_pixel]
+        # average over pulses
+        self.calibration_data = np.mean(self.calibration_data, axis=1)
+        self.calibration_unc = np.mean(self.calibration_unc, axis=0)
+        self.calibration_mask = np.array([True for _ in range(self.calibration_data.shape[0])])
```

Comment:
Nitpicking, an alternative would be:
```suggestion
        self.calibration_mask = np.ones(self.calibration_data.shape[0], dtype=bool)
```